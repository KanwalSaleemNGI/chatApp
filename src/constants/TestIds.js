export default {
  authButton: {
    title: 'title',
    onPress: 'onPress',
  },
  chatInfo: {
    onPress: 'onPress',
    image: 'image',
    name: 'name',
    messageText: 'userChatData',
    icon: 'icon',
    photoText: 'photoText',
    date: 'date',
  },
  login: {
    emailInput: 'emailInput',
    passwordInput: 'passwordInput',
  },
};
