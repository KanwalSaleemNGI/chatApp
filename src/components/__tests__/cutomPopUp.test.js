import React from 'react';
import {render} from '@testing-library/react-native';
import {CustomPopUp} from '..';
import TestIds from '../../constants/TestIds';

describe('customPopUp comp', () => {
  const appTitle = 'chatApp';
  const title = 'Notification';
  const body = 'New message arrived';

  const renderHandler = () => {
    return render(
      <CustomPopUp appTitle={appTitle} title={title} body={body} />,
    );
  };
  it('should renders correctly', () => {
    const {toJSON} = renderHandler();
    expect(toJSON()).toMatchSnapshot();
  });

  it('should properly renders the app title', () => {
    const {getByTestId} = renderHandler();

    expect(getByTestId(TestIds.customPopUp.appTitle).props.children).toEqual(
      appTitle,
    );
  });
  it('should properly renders the title', () => {
    const {getByTestId} = renderHandler();

    expect(getByTestId(TestIds.customPopUp.title).props.children).toEqual(
      title,
    );
  });
  it('should properly renders the body', () => {
    const {getByTestId} = renderHandler();

    expect(getByTestId(TestIds.customPopUp.body).props.children).toEqual(body);
  });
});
