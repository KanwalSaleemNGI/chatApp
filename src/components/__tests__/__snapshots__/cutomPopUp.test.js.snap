// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`customPopUp comp should renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#4C96D6",
      "borderRadius": 15,
      "padding": 15,
      "width": "100%",
    }
  }
>
  <Text
    style={
      Object {
        "fontFamily": "Roboto-Bold",
        "fontSize": 18,
      }
    }
    testID="appTitle"
  >
    chatApp
  </Text>
  <Text
    style={
      Object {
        "fontFamily": "Roboto-Medium",
        "fontSize": 16,
        "marginTop": 5,
      }
    }
    testID="title"
  >
    Notification
  </Text>
  <Text
    style={
      Object {
        "fontFamily": "Roboto-Regular",
        "fontSize": 14,
        "marginTop": 5,
      }
    }
    testID="body"
  >
    New message arrived
  </Text>
</View>
`;
